tests/components/onkyo/test_config_flow.py

```diff
@@ -20,12 +20,60 @@
 
 from . import (
     create_config_entry_from_info,
+    create_connection,
     create_empty_config_entry,
     create_receiver_info,
     setup_integration,
 )
 
-from tests.common import Mock, MockConfigEntry
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+def patch_timeouts():
+    """Patch timeouts to avoid tests waiting."""
+    with (
+        patch(
+            "homeassistant.components.onkyo.receiver.DEVICE_INTERVIEW_TIMEOUT", new=0
+        ),
+        patch(
+            "homeassistant.components.onkyo.receiver.DEVICE_DISCOVERY_TIMEOUT", new=0
+        ),
+    ):
+        yield
+
+
+@pytest.fixture
+async def default_mock_discovery():
+    """Mock discovery with a single device."""
+
+    async def mock_discover(host=None, discovery_callback=None, timeout=0):
+        await discovery_callback(create_connection(1))
```

Comment:
Considered just completely mocking the `Connection` object? You can then call the function via the call arguments that were supplied to the client. It would at least avoid having to implement the function to raise an exception, since you can just set a sideeffect for example.

Not saying that this doesn't work, but maybe have a check at `knocki`

Reply:
Could you clarify what you mean when you said "You can then call the function via the call arguments that were supplied to the client"?  I checked the Knocki tests, but wasn't quite sure what you were referencing.

The connection object here is already a Mock

https://github.com/home-assistant/core/pull/132653/files#diff-7d3bc329d2527df5a442c543f87824564b226d91abba3f0b6a9bab153fb51c3cR22

Reply:
So in Knocki I have
```py
    event_function: Callable[[Event], None] = (
        mock_knocki_client.register_listener.call_args[0][1]
    )

    async def _call_event_function(
        device_id: str = "KNC1-W-00000214", trigger_id: int = 31
    ) -> None:
        event_function(
            Event(
                EventType.TRIGGERED,
                Trigger(
                    device_id=device_id, details=TriggerDetails(trigger_id, "aaaa")
                ),
            )
        )
        await hass.async_block_till_done()
```

This way, I am not creating mock functions in the tests.

If I want the tests to fail when we call `register_listener` I can do `mock_knocki_client.register_listener.side_effect = Exception` and it will raise an exception when called. I personally find this cleaner than creating a method to mock where we do that.

But I am not 100% sure if this is the right way to go

Reply:
Discussed on discord. This approach won't work the same way, since the methods are not called within the test itself. Instead, we’ve just moved the fixtures to the conftest file